File: frontend/src/api/loyalty.ts
import { api } from "./client";

export interface LoyaltyHistoryEntry {
  id: number;
  type: "earn" | "redeem";
  points: number;
  description: string;
  billId: number | null;
  createdAt: string;
}

export interface LoyaltyResponse {
  points: number;
  history: LoyaltyHistoryEntry[];
}

/** Lấy điểm + lịch sử điểm của khách đang đăng nhập (GET /loyalty/me). */
export async function getMyLoyalty(): Promise<LoyaltyResponse> {
  const { data } = await api.get<LoyaltyResponse>("/loyalty/me");
  return data;
}

/** Lấy số điểm của một user cụ thể (dashboard khi xem hoá đơn). */
export async function getUserPoints(userId: number): Promise<number> {
  const { data } = await api.get<{ userId: number; points: number }>(`/loyalty/by-user/${userId}`);
  return data.points;
}

File: backend/Models/Loyalty.cs
namespace MyApi.Models;

public class Loyalty
{
    public int Id { get; set; }

    // 1-1 với User, unique
    public required int UserId { get; set; }
    public User? User { get; set; }

    public int Points { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public List<LoyaltyHistory> History { get; set; } = new();
}